contracts/contracts/layer_0/hold/HoldStorageWrapper2.sol

```diff
@@ -467,13 +488,38 @@ abstract contract HoldStorageWrapper2 is
         delete holdStorage.holdsByAccountPartitionAndId[
             _holdIdentifier.tokenHolder
         ][_holdIdentifier.partition][_holdIdentifier.holdId];
+
+        delete holdStorage.holdThirdParty[_holdIdentifier.tokenHolder][
+            _holdIdentifier.partition
+        ][_holdIdentifier.holdId];
+
         _removeLabafHold(
             _holdIdentifier.partition,
             _holdIdentifier.tokenHolder,
             _holdIdentifier.holdId
         );
     }
 
+    function _restoreAllowance(
+        address _tokenHolder,
+        bytes32 _partition,
+        uint256 _holdId,
+        uint256 _amount
+    ) internal {
```

Comment:
Create in ERC20StorageWrapper2 a new method called _increaseAllowedBalance and remove _restoreAllowance.